src/app/ReadClient.cpp

```diff
@@ -187,6 +163,38 @@ CHIP_ERROR ReadClient::SendReadRequest(NodeId aNodeId, Transport::AdminId aAdmin
     return err;
 }
 
+CHIP_ERROR ReadClient::GenerateAttributePathList(ReadRequest::Builder & aRequest, AttributePathParams * apAttributePathParamsList,
+                                                 size_t aAttributePathParamsListSize)
+{
+    AttributePathList::Builder attributePathListBuilder = aRequest.CreateAttributePathListBuilder();
+    ReturnErrorOnFailure(attributePathListBuilder.GetError());
+    for (size_t index = 0; index < aAttributePathParamsListSize; index++)
+    {
+        AttributePath::Builder attributePathBuilder = attributePathListBuilder.CreateAttributePathBuilder();
+        attributePathBuilder.NodeId(apAttributePathParamsList[index].mNodeId)
+            .EndpointId(apAttributePathParamsList[index].mEndpointId)
+            .ClusterId(apAttributePathParamsList[index].mClusterId);
+        if (apAttributePathParamsList[index].mFlags == AttributePathFlags::kFieldIdValid)
+        {
+            attributePathBuilder.FieldId(apAttributePathParamsList[index].mFieldId);
+        }
+        else if (apAttributePathParamsList[index].mFlags == AttributePathFlags::kFieldIdListIndexValid)
```

Comment:
It's not clear to me why mFlags is `BitFlags` while you seem to use it like an enum. Is it possible to have only the list index without the field id? If yes check both bits independently without the `else` keword. If not, you don't need `BitFlags` and just switch to a bare enum class.

Reply:
agree, just use BitFlags to bit comparison, thanks